testing/e2e/client_api_test.go

```diff
@@ -0,0 +1,219 @@
+			AgentId:  agentID,
+		})
+	}
+
+	resp, err := client.AgentAcksWithResponse(tester.ctx,
+		agentID,
+		&api.AgentAcksParams{},
+		api.AgentAcksJSONRequestBody{
+			Events: events,
+		},
+	)
+	tester.Require().NoError(err)
+	tester.Require().Equal(http.StatusOK, resp.StatusCode())
+
+	acks := resp.JSON200
+	tester.Require().Falsef(acks.Errors, "error in acked items: %v", acks.Items)
+}
+
+func (tester *ClientAPITester) TestFullFileUpload(apiKey, agentID, actionID string, size int64) {
+	client, err := api.NewClientWithResponses(tester.endpoint, api.WithHTTPClient(tester.client), api.WithRequestEditorFn(func(ctx context.Context, req *http.Request) error {
+		req.Header.Set("Authorization", "ApiKey "+apiKey)
+		return nil
+	}))
+	tester.Require().NoError(err)
+
+	beginResp, err := client.UploadBeginWithResponse(tester.ctx,
+		&api.UploadBeginParams{},
+		api.UploadBeginJSONRequestBody{
+			ActionId: actionID,
+			AgentId:  agentID,
+			File: api.UploadBeginRequest_File{
+				Name:     "test-file.zip",
+				MimeType: "application/zip",
+				Size:     size,
+			},
+			Src: api.Agent,
+		},
+	)
+	tester.Require().NoError(err)
+	tester.Require().Equal(http.StatusOK, beginResp.StatusCode())
+	tester.Require().NotEmpty(beginResp.JSON200.UploadId)
+	tester.Require().NotZero(beginResp.JSON200.ChunkSize)
+	uploadID := beginResp.JSON200.UploadId
+	chunkSize := beginResp.JSON200.ChunkSize
+
+	chunkCount := int(math.Ceil(float64(size) / float64(chunkSize)))
+	tHash := sha256.New()
+	for i := 0; i < chunkCount; i++ {
+		var body bytes.Buffer
+		_, err := io.CopyN(&body, rand.Reader, chunkSize)
+		tester.Require().NoError(err)
+		hash := sha256.Sum256(body.Bytes())
+		chunkResp, err := client.UploadChunkWithBodyWithResponse(tester.ctx,
+			uploadID,
+			i,
+			&api.UploadChunkParams{XChunkSHA2: fmt.Sprintf("%x", hash[:])},
+			"application/octet-stream",
+			&body,
+		)
+		tester.Require().NoError(err)
+		tester.Require().Equal(http.StatusOK, chunkResp.StatusCode())
+		tHash.Write(hash[:])
+	}
+
+	completeResp, err := client.UploadCompleteWithResponse(tester.ctx,
+		uploadID,
+		&api.UploadCompleteParams{},
+		api.UploadCompleteJSONRequestBody{
+			Transithash: struct {
+				Sha256 string `json:"sha256"`
+			}{
+				Sha256: fmt.Sprintf("%x", tHash.Sum(nil)),
+			},
+		},
+	)
+	tester.Require().NoError(err)
+	tester.Require().Equal(http.StatusOK, completeResp.StatusCode())
+}
+
+func (tester *ClientAPITester) TestArtifact(apiKey, id, sha2 string) {
+	client, err := api.NewClientWithResponses(tester.endpoint, api.WithHTTPClient(tester.client), api.WithRequestEditorFn(func(ctx context.Context, req *http.Request) error {
+		req.Header.Set("Authorization", "ApiKey "+apiKey)
+		return nil
+	}))
+	tester.Require().NoError(err)
+
+	resp, err := client.ArtifactWithResponse(tester.ctx,
+		id,
+		sha2,
+		&api.ArtifactParams{},
+	)
+	tester.Require().NoError(err)
+	tester.Require().Equal(http.StatusOK, resp.StatusCode())
+	// TODO more validation?
```

Comment:
would like some feedback here, or if just ensuring that the call returns a 200 is enough

Reply:
Can we test that the body contains the proper artifact? maybe comparing the sha256 as we already have it here?

Reply:
comparing with the encoded hash